.ci/test-install-docs.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+# Detect if any installation guides changed. If so, run the
+# kata manager to "execute" the install guide to ensure the
+# commands it specified result in a working system.
+check_install_guides()
+{
+	local -r install_guide_suffix="-installation-guide.md"
+
+	# List of filters used to restrict the types of file changes.
+	# See git-diff-tree(1) for further info.
+	local filters=""
+
+	# Added file
+	filters+="A"
+
+	# Copied file
+	filters+="C"
+
+	# Modified file
+	filters+="M"
+
+	# Renamed file
+	filters+="R"
+
+	# Unmerged (U) and Unknown (X) files. These particular filters
+	# shouldn't be necessary but just in case...
+	filters+="UX"
+
+	# List of changed files
+	local files=$(git diff-tree \
+		--name-only \
+		--no-commit-id \
+		--diff-filter="${filters}" \
+		-r \
+		origin/master HEAD || true)
+
+	# No files were changed
+	[ -z "$files" ] && return
+
+	changed=$(echo "$files" | grep "install/.*${install_guide_suffix}$" || true)
+
+	# None of the install guides were modified
+	[ -z "$changed" ] && return
+
+	echo "INFO: Found modified install guides: $changed"
+
+	# Regardless of which distro install guide(s) were changed, we test
+	# them all where possible.
+
+	local -r GOPATH=$(go env GOPATH)
+	[ -z "$GOPATH" ] && die "cannot determine GOPATH"
+
+	local -r mgr="${GOPATH}/src/github.com/kata-containers/tests/cmd/kata-manager/kata-manager.sh"
+
+	[ ! -e "$GOPATH" ] && die "cannot find $mgr"
+
+	source /etc/os-release
+
+	echo "INFO: Installing system from the $ID install guide"
+
+	$mgr install-packages
```

Comment:
I'm a little confused how `$mgr` relates to `$ID`, and then if/how that has any affect/interaction/relationship to `$changed` - maybe a comment here giving a quick overview of the relationship would be good (as it is not clear to me from just the code that if `$changed` does not include the file relevant to `$ID` - do we do anything useful, and would we check the file that changed iyswim).
I know, it is partly my ignorance for not following the details of how the auto-docs scripts work - but I'm sure I'm not the only one ;-)

Reply:
I'm not quite clear on what the problem is I'm afraid?

There are two comments in the test that explain that if *any* install guide changes, we re-test *all* the install guides (why not? :) The script is simply using `$ID` for informational purposes - the `$mgr` script will dtrt and install the packages for the distro the script is called on (or error if it cannot find an install doc for the current distro).

Reply:
I know it says 'all the scripts', and then the script itself says `INFO: Installing system from the $ID install guide`. I just didn't, obviously, see how 'all' fitted in with that specific looking echo. np - as I said, not obvious to me as I don't know exactly how this ties into the scripts it is calling.

Reply:
Oh right. What I was trying to get as was, since this is a CI script, it will run on 'n' distros and if the install guide for distro 'a' is modified, we'll still re-check all 'n' distros (by running this new CI script once on each of the 'n' distros our CI is configured for). This seems reasonable given the jobs run in parallel and it doesn't hurt for us to re-test as much as possible on all distros given package versions (and their dependent package versions) may change frequently.